packages/server/modules/core/repositories/users.ts

```diff
@@ -449,39 +450,84 @@ export const getUserRoleFactory =
   }
 
 /**
- * User search available for normal server users. It's more limited because of the lower access level.
+ * Used for (Limited)User search. No need to convert users to Limited here, because non-limited fields
+ * cannot be leaked out from the GQL API.
  */
-export const searchUsersFactory =
-  (deps: { db: Knex }): SearchLimitedUsers =>
-  async (searchQuery, limit, cursor, archived = false, emailOnly = false) => {
-    const prefixedLimitedUserFields = LIMITED_USER_FIELDS.map(
-      (field) => `users.${field}`
-    )
+export const lookupUsersFactory =
+  (deps: { db: Knex }): LookupUsers =>
+  async (filter) => {
+    const {
+      query: searchQuery,
+      limit = 10,
+      cursor,
+      archived = false,
+      emailOnly = false,
+      projectId
+    } = filter
+
     const query = tables
       .users(deps.db)
-      .join('server_acl', 'users.id', 'server_acl.userId')
+      .join(ServerAcl.name, Users.col.id, ServerAcl.col.userId)
       .leftJoin(UserEmails.name, UserEmails.col.userId, Users.col.id)
       .columns([
-        ...Object.values(omit(Users.col, ['email', 'verified'])).filter((col) =>
-          prefixedLimitedUserFields.includes(col)
+        ...Object.values(
+          omit(Users.col, [
+            Users.col.email,
+            Users.col.verified,
+            Users.col.passwordDigest
+          ])
         ),
-        knex.raw(`(array_agg("user_emails"."verified"))[1] as verified`)
+        knex.raw(`(array_agg(??))[1] as "verified"`, [UserEmails.col.verified]),
```

Comment:
@alemagio it was this way before already, but i'm wondering if this logic is valid - surely we should look for the first verified email, instead of just the first email that exists? otherwise we risk marking the user as unverified, when one of their emails is verified